src/Domain/Insights/InsightCollectionFactory.php

```diff
@@ -59,17 +59,17 @@ public function get(
         array $metrics,
         OutputInterface $consoleOutput
     ): InsightCollection {
-        $dir = $this->config->getDirectory();
+        $directoryList = $this->config->getDirectory();
 
         try {
             $files = array_map(static function (\SplFileInfo $file) {
                 return $file->getRealPath();
-            }, iterator_to_array($this->filesRepository->within($dir, $this->config->getExcludes())->getFiles()));
+            }, $this->filesRepository->within($directoryList, $this->config->getExcludes())->getFiles());
         } catch (\InvalidArgumentException $exception) {
             throw new DirectoryNotFound($exception->getMessage(), 0, $exception);
         }
 
-        $collector = $this->analyser->analyse($dir, $files);
+        $collector = $this->analyser->analyse($directoryList[0], $files);
```

Comment:
Here we have again a scenario where we have to figure out the approach.
The dir supplied here is passed down to the `Collector`, in the collector it is used to shorten down the name of the file to only filename and not the full path.